novawallet/Modules/Staking/NominationPools/BondMore/Setup/NominationPoolBondMorePresenter.swift

```diff
@@ -0,0 +1,182 @@
+            return nil
+        }
+        switch inputResult {
+        case .rate:
+            let fee = fee?.decimal(precision: chainAsset.asset.precision) ?? 0
+            let availableAmountDecimal = transferrableBalance ?? 0
+            return inputResult.absoluteValue(from: availableAmountDecimal - fee)
+        case let .absolute(inputValue):
+            return inputResult.absoluteValue(from: inputValue)
+        }
+    }
+
+    override func updateView() {
+        provideAssetViewModel()
+        provideAmountInputViewModel()
+        provideTransferrableBalance()
+        provideFee()
+        provideHints()
+    }
+
+    override func provideHints() {
+        let hints = hintsViewModelFactory.createHints(
+            rewards: claimableRewards,
+            locale: selectedLocale
+        )
+        view?.didReceiveHints(viewModel: hints)
+    }
+
+    override func provideFee() {
+        guard let fee = fee?.decimal(precision: chainAsset.asset.precision) else {
+            view?.didReceiveFee(viewModel: nil)
+            return
+        }
+
+        let viewModel = balanceViewModelFactory.balanceFromPrice(fee, priceData: price).value(for: selectedLocale)
+
+        view?.didReceiveFee(viewModel: viewModel)
+    }
+
+    override func getInputAmountInPlank() -> BigUInt? {
+        guard let decimalAmount = getInputAmount() else {
+            return nil
+        }
+
+        return decimalAmount.toSubstrateAmount(
+            precision: chainAsset.assetDisplayInfo.assetPrecision
+        )
+    }
+
+    override func didReceive(assetBalance: AssetBalance?) {
+        super.didReceive(assetBalance: assetBalance)
+
+        transferrableBalance = assetBalance.map {
+            Decimal.fromSubstrateAmount(
+                $0.transferable,
+                precision: Int16(chainAsset.asset.precision)
+            ) ?? 0
+        }
+
+        provideAssetViewModel()
+        provideTransferrableBalance()
+        provideAmountInputViewModel()
+    }
+
+    override func didReceive(stakingLedger: StakingLedger?) {
+        super.didReceive(stakingLedger: stakingLedger)
+
+        provideAssetViewModel()
+        provideAmountInputViewModel()
+    }
+
+    override func didReceive(price: PriceData?) {
+        super.didReceive(price: price)
+
+        provideAssetViewModel()
+        provideAmountInputViewModel()
+        provideTransferrableBalance()
+        provideFee()
+    }
+}
+
+extension NominationPoolBondMorePresenter: NominationPoolBondMorePresenterProtocol {
+    func setup() {
+        updateView()
+        interactor?.setup()
+    }
+
+    func selectAmountPercentage(_ percentage: Float) {
+        inputResult = .rate(Decimal(Double(percentage)))
+
+        provideAssetViewModel()
+        provideAmountInputViewModel()
+        refreshFee()
+    }
+
+    func updateAmount(_ newValue: Decimal?) {
+        inputResult = newValue.map { AmountInputResult.absolute($0) }
+        refreshFee()
```

Comment:
we should also call provideAssetViewModel because with changing amount price also changes